src/main/java/org/elasticsearch/index/query/ScriptQueryParser.java

```diff
@@ -137,23 +138,27 @@ public String toString(String field) {
         }
 
         @Override
-        public boolean equals(Object o) {
-            if (this == o) return true;
-            if (super.equals(o) == false) return false;
-
-            ScriptQuery that = (ScriptQuery) o;
-
-            if (params != null ? !params.equals(that.params) : that.params != null) return false;
-            if (script != null ? !script.equals(that.script) : that.script != null) return false;
-
+        public boolean equals(Object obj) {
+            if (this == obj)
+                return true;
+            if (!super.equals(obj))
+                return false;
+            if (getClass() != obj.getClass())
+                return false;
+            ScriptQuery other = (ScriptQuery) obj;
+            if (script == null) {
+                if (other.script != null)
+                    return false;
+            } else if (!script.equals(other.script))
+                return false;
```

Comment:
Maybe let's just call Objects.equal(script, other.script) for simplicity? I know you did not introduce it though...